New Entry: C++ Templates (#879)
Codecademy/docs: content/cpp/concepts/templates/templates.md

---
Title: 'Templates'
Description: 'Templates provide the ability to use a type as a parameter in functions and classes.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Classes'
  - 'Objects'
  - 'Data Types'
  - 'Constructors'
  - 'Methods'
  - 'Functions'
  - 'OOP'

>   - 'Parameters'

Reviewer: ```suggestion
  - 'Parameters'
  - 'Templates'
```

CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> **Templates** provide the ability to use a [type](https://www.codecademy.com/resources/docs/cpp/data-types) as a parameter in [functions](https://www.codecademy.com/resources/docs/cpp/functions) and [classes](https://www.codecademy.com/resources/docs/cpp/classes). Such classes and functions are referred to as using a generic type. This provides the ability to define a set of related classes or functions that can operate on many different types with a single declaration.

Reviewer: ```suggestion
**Templates** provide the ability to use a [data type](https://www.codecademy.com/resources/docs/cpp/data-types) as a parameter in [functions](https://www.codecademy.com/resources/docs/cpp/functions) and [classes](https://www.codecademy.com/resources/docs/cpp/classes). These are referred to as generic types. This provides the ability to define a set of related classes or functions that can operate on many different types with a single declaration.
```
